integrations/qdrant/src/haystack_integrations/document_stores/qdrant/converters.py

```diff
@@ -18,24 +21,23 @@ def documents_to_batch(
         documents: List[Document],
         *,
         embedding_field: str,
+        sparse_embedding_field: str,
     ) -> List[rest.PointStruct]:
         points = []
         for document in documents:
             payload = document.to_dict(flatten=False)
-            vector = payload.pop(embedding_field) or {}
+            vector = {}
+            if embedding_field in payload and payload[embedding_field] is not None:
+                dense_vector = payload.pop(embedding_field) or []
+                vector[DENSE_VECTORS_NAME] = dense_vector
+            # TODO: does this work with this https://github.com/deepset-ai/haystack/pull/7382#discussion_r1530515595 ?
+            # TODO: maybe check if not empty string also on top of None ?
+            if sparse_embedding_field in payload and payload[sparse_embedding_field] is not None:
+                sparse_vector = payload.pop(sparse_embedding_field, {"indices": [], "values": []})
+                sparse_vector_instance = rest.SparseVector(**sparse_vector)
+                vector[SPARSE_VECTORS_NAME] = sparse_vector_instance
```

Comment:
```suggestion
            sparse_vector = payload.pop(sparse_embedding_field, {"indices": [], "values": []})
            sparse_vector_instance = rest.SparseVector(**sparse_vector)
            vector[SPARSE_VECTORS_NAME] = sparse_vector_instance
```

Since you are providing a default for pop, this should work.

Reply:
```python
            # TODO: does this work with this https://github.com/deepset-ai/haystack/pull/7382#discussion_r1530515595 ?
            # TODO: maybe check if not empty string also on top of None ?
            sparse_vector = payload.pop(sparse_embedding_field, {"indices": [], "values": []})
>           sparse_vector_instance = rest.SparseVector(**sparse_vector)
E           TypeError: qdrant_client.http.models.models.SparseVector() argument after ** must be a mapping, not NoneType

src/haystack_integrations/document_stores/qdrant/converters.py:34: TypeError
```

Reply:
ok you are right.
let's add a check somewhere that raises an error if `sparse_embedding_field` has 0 length.
The same for `embedding_field`

Reply:
> ok you are right. let's add a check somewhere that raises an error if `sparse_embedding_field` has 0 length. The same for `embedding_field`

Sooooooo I reverted it to this and test are good:

```python
            if embedding_field in payload and payload[embedding_field] is not None:
                dense_vector = payload.pop(embedding_field) or []
                vector[DENSE_VECTORS_NAME] = dense_vector
            if (
                sparse_embedding_field in payload
                and payload[sparse_embedding_field] is not None
                and payload[sparse_embedding_field] != ""
            ):
                sparse_vector = payload.pop(sparse_embedding_field, {"indices": [], "values": []})
                sparse_vector_instance = rest.SparseVector(**sparse_vector)
                vector[SPARSE_VECTORS_NAME] = sparse_vector_instance
```
But I guess maybe you don't like the `and payload[sparse_embedding_field] != ""` check ?

Also I'm still wondering what to do with this: (put to none or empty string ?)
```
class QdrantToHaystack:
    def __init__(self, content_field: str, name_field: str, embedding_field: str, sparse_embedding_field: str):
        self.content_field = content_field
        self.name_field = name_field
        self.embedding_field = embedding_field
        self.sparse_embedding_field = sparse_embedding_field

    def point_to_document(self, point: QdrantPoint) -> Document:
        payload = {**point.payload}
        if hasattr(point, "vector") and point.vector is not None and DENSE_VECTORS_NAME in point.vector:
            payload["embedding"] = point.vector[DENSE_VECTORS_NAME]
        else:
            payload["embedding"] = None
        payload["score"] = point.score if hasattr(point, "score") else None
        if hasattr(point, "vector") and point.vector is not None and SPARSE_VECTORS_NAME in point.vector:
            parse_vector_dict = {
                "indices": point.vector[SPARSE_VECTORS_NAME].indices,
                "values": point.vector[SPARSE_VECTORS_NAME].values,
            }
            payload["sparse_embedding"] = parse_vector_dict
        else:
            # TODO: does this work with this https://github.com/deepset-ai/haystack/pull/7382#discussion_r1530515595 ?
            payload["sparse_embedding"] = None
        return Document.from_dict(payload)
```

Reply:
> In Document Store **init**, I would raise an error if embedding_field or sparse_embedding_field are ""

I just did an end-to-end test on my haystack app and everything is working fine now.
If  you want to add a raised error can you do it ? I think we are not talking about the same thing and I'm not sure I understand what you want to check :)
Your concerns are about the name of fields being empty
My concerns were about the content of document's sparsembedding being an empty string (and not none) as defined in the dataclass (trick to not change the existing IDs https://github.com/deepset-ai/haystack/pull/7382#discussion_r1530515595)

Reply:
I simplified a bit this function.

A bit unrelated: I see several things that can be simplified in this Document Store (not related to the Sparse feature)